### DISCUSSION TITLE
Running both Initiator and Acceptor in Openshift

### DISCUSSION THREAD

**Discussion by DeonBoon** [quickfix-j/quickfixj#476] [Q&A]
Hi! Thank for this awesome project and I have a few questions to ask regarding the running of both Initiator and Acceptor in Openshift/Kubernetes environments.

I have developed the Initiator and Acceptor and they are able to communicate and send messages back and forth locally using this configuration:
```
Initiator
SocketConnectPort=15000
SocketConnectHost=localhost

Acceptor
SocketAcceptPort=15000
```
When running in Openshift, I have created the routes for my acceptor for both HTTP and HTTPS traffic with port forwarding pointing to 15000 in my acceptor service with these sets of configurations:
```
#HTTP 
Initiator
SocketConnectPort=80
SocketConnectHost=openshift.com

Acceptor
SocketAcceptPort=15000
```

```
#HTTPS
Initiator
SocketConnectPort=443
SocketConnectHost=openshift2.com

Acceptor
SocketAcceptPort=15000
```
I ran the above configurations separately but both encountered these errors:
```
Initiator -> IO Exception, Disconnecting, END_OF_STREAM
Acceptor -> Connection reset by peer
```
However, If i configure my initiator to point directly to the acceptor service instead of openshift route, the connection is able to establish successfully.
```
Initiator
SocketConnectPort=15000
SocketConnectHost=acceptor
```
May I ask if anyone has any idea what should be the correct configuration for the routes?

**Comment by chrjohn** [comment]
I am not familiar with OpenShift but it sounds like a route will only accept HTTP(S) traffic. But without wrapping FIX into HTTP it probably won't work. Maybe you could try the http proxy configuration options?
